src/3D/MeshPack.cpp

```diff
@@ -34,20 +37,78 @@ MeshPack::MeshPack(bool enableUnknownMeshes)
 bool MeshPack::LoadFromFile(const fs::path& path)
 {
 	SPDLOG_LOGGER_DEBUG(spdlog::get("game"), "Loading Mesh Pack from file: {}", path.generic_string());
-	pack::PackFile pack;
+	auto pathExt = path.extension().string();
+	LowerCase(pathExt);
 
 	try
 	{
-		pack.Open(Game::instance()->GetFileSystem().FindPath(path).u8string());
+		if (pathExt == ".g3d")
+		{
+			pack::PackFile pack;
+			pack.Open(Game::instance()->GetFileSystem().FindPath(path).u8string());
+			loadTextures(pack.GetTextures());
+			loadMeshes(pack.GetMeshes());
+			return true;
+		}
 	}
 	catch (std::runtime_error& err)
 	{
 		SPDLOG_LOGGER_ERROR(spdlog::get("game"), "Failed to open {}: {}", path.generic_string(), err.what());
 		return false;
 	}
 
-	loadTextures(pack.GetTextures());
-	loadMeshes(pack.GetMeshes());
+	return false;
+}
+
+std::pair<bool, MeshId> MeshPack::LoadLooseMeshFile(const fs::path& path)
+{
+	SPDLOG_LOGGER_DEBUG(spdlog::get("game"), "Loading Mesh Pack from file: {}", path.generic_string());
+	auto pathExt = path.extension().string();
+	LowerCase(pathExt);
```

Comment:
You're already making a copy of the string by using `auto`. Also, keeping the namespace makes the code more readable.
This is why I think it's better to just do this:
```c++
const auto pathExt = StringUtils::LowerCase(path.extension().string());
```

Reply:
Wouldn't `namespace openblack::string` be better than `StringUtils`?

Reply:
Usually, you call it string if you're making your own implementation.